src/server/graphql/mutations/stripeCreateInvoice.js

```diff
@@ -0,0 +1,33 @@
+import fetchAllLines from 'server/billing/helpers/fetchAllLines';
+import generateInvoice from 'server/billing/helpers/generateInvoice';
+import stripe from 'server/billing/stripe';
+import resolvePromiseObj from 'universal/utils/resolvePromiseObj';
+import {GraphQLBoolean, GraphQLID, GraphQLNonNull} from 'graphql';
+
+export default {
+  name: 'CreateInvoice',
+  description: 'When stripe tells us an invoice is ready, create a pretty version',
+  type: GraphQLBoolean,
+  args: {
+    invoiceId: {
+      type: new GraphQLNonNull(GraphQLID),
+      description: 'The stripe invoice ID'
+    }
+  },
+  resolve: async (source, {invoiceId}, {serverSecret}) => {
+    // AUTH
+    if (serverSecret !== process.env.AUTH0_CLIENT_SECRET) {
```

Comment:
our server has a secret. it's the same secret that auth0 uses to sign JWTs. If anyone ever found that secret, we'd be screwed.
The only entity that should ever call this mutation is our own server. If anyone else called it, then we'd be screwed. So, the only way to make sure that our server is calling itself is to ask the server something only it would know.

Reply:
I understand the need for a server secret. We could simply generate a new secret for this purpose, so that if our `AUTH0_CLIENT_SECRET` were compromised, we wouldn't be screwed for these non-auth0 use cases.

Reply:
if the AUTH0 secret gets compromised, it's game over. they could access our GraphQL endpoint directly, sign their own JWT to impersonate me or anyone else in the entire app, etc. However, if it gives you tension we could create a webhook secret & that'll be a good excuse to show ya how to update env variables across deployments